src/main/java/bio/terra/service/dataset/AssetDao.java

```diff
@@ -89,7 +88,7 @@ public void retrieve(Dataset dataset) {
 
     // also retrieves dependent objects
     public List<AssetSpecification> retrieveAssetSpecifications(Dataset dataset) {
-        Map<UUID, Table> allTables = dataset.getTablesById();
+        Map<UUID, DatasetTable> allTables = dataset.getTablesById();
```

Comment:
This is a theme in multiple files, it's almost entirely plumbing to make sure that the PDAO can get access to dataset-specific metadata when it needs to.

Reply:
Makes sense.